pytorch_lightning/callbacks/progress/base.py

```diff
@@ -159,6 +159,8 @@ def print(self, *args, **kwargs):
 
     def setup(self, trainer: "pl.Trainer", pl_module: "pl.LightningModule", stage: Optional[str] = None) -> None:
         self._trainer = trainer
+        if not trainer.is_global_zero:
```

Comment:
let's do this to be explicit about enable/disable
```suggestion
        if trainer.is_global_zero:
            self.enable()
        else:
            self.disable()
```

Reply:
I had this initially, but this will break the case where we manually disable the progress bar before the setup hook gets called, like in this test: https://github.com/PyTorchLightning/pytorch-lightning/blob/3e0569fccc9808b1017ace63951485e2f76563eb/tests/callbacks/test_tqdm_progress_bar.py#L465-L480

If we remove `enable` and `disable` from the progress bar callback API this will no longer be an issue, but we would have to wait for the dep. cycle